.gitignore

```diff
@@ -1,31 +1,121 @@
-# Compiled Object files, Static and Dynamic libs (Shared Objects)
-*.o
-*.a
+# don't accidentally commit a build of main
+boilr
+
+# Created by https://www.toptal.com/developers/gitignore/api/go,intellij+iml
+# Edit at https://www.toptal.com/developers/gitignore?templates=go,intellij+iml
+
+### Go ###
+# If you prefer the allow list template instead of the deny list, see community template:
+# https://github.com/github/gitignore/blob/main/community/Golang/Go.AllowList.gitignore
+#
+# Binaries for programs and plugins
+*.exe
+*.exe~
+*.dll
 *.so
+*.dylib
 
-# Folders
-_obj
-_test
+# Test binary, built with `go test -c`
+*.test
 
-# Architecture specific extensions/prefixes
-*.[568vq]
-[568vq].out
+# Output of the go coverage tool, specifically when used with LiteIDE
+*.out
 
-*.cgo1.go
-*.cgo2.c
-_cgo_defun.c
-_cgo_gotypes.go
-_cgo_export.*
+# Dependency directories (remove the comment below to include it)
+# vendor/
 
-_testmain.go
+# Go workspace file
+go.work
 
-*.exe
-*.test
-*.prof
+### Go Patch ###
+/vendor/
+/Godeps/
 
-# dep management files
-glide.lock
-vendor
+### Intellij+iml ###
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio, WebStorm and Rider
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
 
-# don't accidentally commit a build of main
-boilr
+# User-specific stuff
+.idea/**/workspace.xml
+.idea/**/tasks.xml
+.idea/**/usage.statistics.xml
+.idea/**/dictionaries
+.idea/**/shelf
```

Comment:
I'm a little confused why we are adding all this to the gitignore?

Reply:
Ohhh I'm using intellij idea as IDE and it always creates this folder, and I seem that other devs use goLand, so I wanted to add the recommended ignores for jetBreains idea. does wattpad has a standard ignore file for go projects?

Reply:
I would just use ` .idea ` - you can look at the highlander gitignore for reference

Reply:
sure, I'll do that.